test/unit/io/sequence_file/sequence_file_format_test_template.hpp

```diff
@@ -0,0 +1,246 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#include <sstream>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/alphabet/quality/all.hpp>
+#include <seqan3/io/sequence_file/all.hpp>
+#include <seqan3/range/views/convert.hpp>
+#include <seqan3/std/ranges>
+#include <seqan3/std/concepts>
+#include <seqan3/test/pretty_printing.hpp>
+
+using namespace seqan3;
+
+struct sequence_file_data : public ::testing::Test
+{
+    std::vector<std::string> ids
+    {
+        { "ID1" },
+        { "ID2" },
+        { "ID3 lala" },
+    };
+
+    std::vector<dna5_vector> seqs
+    {
+        { "ACGTTTTTTTTTTTTTTT"_dna5 },
```

Comment:
so every testable format has to match now exactly these inputs from the test data?
So I would propose making them constants and have the formats as well as this test data fixture use these constants. That makes adding a test much easier for the future.

Reply:
No, since a test fixture does not need to inherit from `sequence_file_data` but may specify these variables itself. So this is flexible but still reduces code for the ones that can be tested with this because they do not need to defines the same thing again

Reply:
I still don't get it. 
>So this is flexible but still reduces code for the ones that can be tested with this because they do not need to defines the same thing again

A format has to be created to match these ids and sequences which is then read from in the public file based interface, right? So there need to be "file" definitions for exactly these values. And they have to be the same. So changing something here. For example, by realising that there was a bug within the qualities or something means I have to go through all the files, figure out which one is using these values and then adapt them by hand there as well. I suggest that this can be circumvented by having global values for these, which can be reused by the tests that want to use them. Of course if they overwrite their behavior then it is specific to the test. But I believe this is a critical aspect here.